The Toiletbong and other poetical additions (*click* Noice)

LordVollkorn opened the pull request:
## About The Pull Request (In a nutshell)
This adds the following content to the game:
- The toiletbong
- The toiletbong craftingrecipe
- Plums
- Plum seeds
- A mutation of plums, called plumbs (also with seeds)
- Plum juice
- Plum wine
- A drink called "The Hat", which is a plum swimming in a man's hat
- Plum cake
- Plum cake slices
- The plum cake recipe

## About The Pull Request (Full)
![hydroponicsafterhours](https://user-images.githubusercontent.com/66637090/177220143-b92a0445-f02a-4727-a7b4-4c2c96b59206.png)

**What this is about**
8 months ago I had the idea of bringing the toilet bong concept into the game, made some sprites for it and [posted the above image on reddit](https://www.reddit.com/r/SS13/comments/qm3bjr/botanist_ok_hear_me_out_on_this_one/). I never got around to coding it, so instead the idea was picked by Yogstation (I gave my permission for the sprites) and they got the first [toilet bong](https://github.com/yogstation13/Yogstation/pull/12605).

I finally decided to code it for /tg/ and initially ported it from Yogstation, but then took a different approach and re-did the thing.
On /tg/ it is built using the craftingmenu, while having a (filled) flamethrower and a wrench nearby.
**It can be rotated with ALT+leftclick while unsecured.
It can be unsecured with a wrench.
It's inventory can be accessed with rightclick or ALT+leftclick (standard for containers).
It can then be filled with various grown/food items.
Using the toiletbong will create a smoke cloud with a radius of 1, spreading the reagents of one item at a time.**

There is a 5% chance that you suck a mouse through the pipes, which is disgusting for most members of the crew. Although some species will see it as a hidden treat.

The toiletbong has an open flame during usage and will ignite plasma in the air (like candles, matches, lighters).

The toiletbong can be emagged for the following effects:
- All contents are consumed and turned into smoke at the same time instead of one by one
- The smoke radius is increased, from 1 to 2
- _You drop your emag into the toilet and fish it back out_

To avoid people stealing the flamethrower (because fires are annoying) deconstructing the toiletbong will return the flamethrower with an empty tank.
This also means that you can ask a nice engineer to build you a toiletbong, and they don't have to worry about you burning down the station.
![bong1](https://user-images.githubusercontent.com/66637090/177220836-ae7aaf0e-7d90-48b6-b8b6-d5ae7f80c5b1.png)
![bong2](https://user-images.githubusercontent.com/66637090/177220838-945a0b95-fa41-4f63-a01f-8977f4aed746.png)
![bong3_gif](https://user-images.githubusercontent.com/66637090/177220839-7a6e128f-a0b9-4d11-a33f-4f42edc7395c.gif)

**Plums, click Noice**

(In case you don't know, this is a reference to [this)](https://www.youtube.com/watch?v=UBX8MWYel3s)

Rosenol is my favorite chemical. It's also based on my favorite meme. Back then, when I created Rosenol, I was asked why I didn't add plums. Well, I sure do love a noice bag of plums, so here we are.
![grown](https://user-images.githubusercontent.com/66637090/177221079-31efa2b4-0210-4309-97b3-7ac083e1fa3b.png)
![food](https://user-images.githubusercontent.com/66637090/177221082-12df70bf-1534-4b8c-a88a-a5e42318488d.png)
![drinks](https://user-images.githubusercontent.com/66637090/177221081-d9712273-a1cc-4808-803d-caaa98426adb.png)

Plum seeds can be bought at the vendor, and be grown into plum trees.
Plums contain small traces of Rosenol, which is **click** Noice.
They can be mutated into **Plumbs** (notice the b).
Plumbs contain Pb, which is latin for Plumbum, meaning Lead. They produce small traces of Lead.

Normal plums can be juiced and fermented.
They may also be put onto a cake, which can be sliced.
Fermenting plums will yield plum wine.

Pouring Ethanol, Water and Plum Wine into a glass, will create the new drink "The Hat".
(This is a reference to [this](https://www.reddit.com/r/WeWantPlates/comments/6g48h9/a_single_plum_floating_in_perfume_served_in_a/))

## Why It's Good For The Game

The toiletbong is a joke item. My goal, first and foremost, is to help create some funny stories and make people laugh when they read the wiki and see all the crazy possibilities in our code base.

The plums (and even "The Hat") are fun additions, not too far out, but still something to laugh about. I believe that chefs and bartenders might enjoy trying those new recipes.

## ToDo if this gets merged
If this gets merged I will add the following to the wiki:
- Toiletbong entry
- Plum/Plumb seed entry
- Plum/Plumb stats entry
- Drinks entry
- Food entry

## Feedback needed
There surely must be something I missed or could have done better while coding this, please point out bits of code that i could improve.

## Changelog

:cl: LordVollkorn / Gage Gaebler
add: The toiletbong has been added to the game. Build it using the craftingmenu, a wrench, a flamethrower and a nearby toilet.
balance: Deconstructing a toiletbong will empty the flamethrower's plasma tank. Engineers, you may build toiletbongs for anyone without feeling guilty.
add: The toiletbong can be emagged. But should you?
add: Plums have been added to Botany. Get your plum seeds today. *Click* Noice.
add: Plumcake and plumcake slices have been added. Find them in the craftingmenu.
add: Plumjuice and plumwine have been added. Pour yourself a glass.
add: A new drink called "The Hat" has been added. A hat is also where you drink it from.
/:cl:

LordVollkorn committed:
Main

LordVollkorn committed:
Merge branch 'master' of https://github.com/tgstation/tgstation.git into poetryUpdate

optimumtact commented:
>To avoid people stealing the flamethrower (because fires are annoying) I did not add any deconstruction routine.
it needs to be deconstructable

optimumtact reviewed (dismissed):
Add deconstruction steps

LordVollkorn commented:
> Add deconstruction steps

Alright then, I will figure something out

LemonInTheDark commented:
why are you adding plumbs in the same pr as toilet bongs

tf-4 commented:
alt+rclick is a bad keybind for rotate, just make it alt click when unsecured

MrDoomBringer commented:
Stoner botanists grabbing a fucking flamethrower off of their DIY toilet gigabongs to ward off assistants sounds absolutely hilarious. Good PR

wesoda25 commented:
Fucking legend

LordVollkorn committed:
Added deconstruction and better rotation

LordVollkorn commented:
> Add deconstruction steps

Deconstruction has been added. Using a crowbar on an unsecured toiletbong will yield metal (from the toilet), a flamethrower and an empty plasma tank.

LordVollkorn commented:
> why are you adding plums in the same pr as toilet bongs

It was originally planned as a "stoner update" and the "plumbs" (mutation of plums) should have been a reference to "plumbing", which relates to the toilet (since you screw with the plumbing in the process). I thought plumbs could yield fluid ducts or plumbing constructors, but then I realized this would be a bit too much.

Before I split this into two separate PRs, I would say that this is one "Botany Update".
_I'd split it if being told to do so_



> alt+rclick is a bad keybind for rotate, just make it alt click when unsecured

You are right. I added some code to take care of this and it should now work as you expect

LordVollkorn committed:
Open flame during usage, emagging

LordVollkorn commented:
Added an open flame during usage, so it will ignite plasma in the air.
Also had the idea of adding emag capabilities (all contents of the toiletbong will be turned into smoke at once, instead of one at a time. Also increases smoke radius from 1 to 2)

LordVollkorn committed:
Wording fix, sound fix

LordVollkorn committed:
Extra-indestructable check

LordVollkorn commented on code/game/objects/structures/toiletbong.dm:
```diff
@@ -0,0 +1,106 @@
+				if(prob(50))
+					for(var/j in 1 to rand(1, 3))
+						step(spawned_mob, pick(NORTH,SOUTH,EAST,WEST))
+			qdel(item)
```
Under normal circumstances I think no, since

```
	STR.set_holdable(list(/obj/item/food/))
```

but to cover all our bases, I would add this here:

```
if (item.resistance_flags & INDESTRUCTIBLE)
				  user.balloon_alert(user, "[item.name] is blocking the pipes!")
				  continue
```

It skips the indestructable item and therefore not delete it.
I tested it and it works in emagged and non-emagged state.

I'll push the changes, let me know what you think 👍

LordVollkorn committed:
Merge branch 'master' of https://github.com/tgstation/tgstation.git into poetryUpdate

LordVollkorn committed:
Storage is now a normal datum instead of a component? Noice

LordVollkorn commented:
Updated and tested to function with the new atom_storage according to [67478](https://github.com/tgstation/tgstation/pull/67478)

LordVollkorn committed:
Merge branch 'master' of https://github.com/tgstation/tgstation.git into poetryUpdate

# Conflicts:
#	icons/obj/hydroponics/harvest.dmi

LordVollkorn committed:
Updated harvest.dmi after bell pepper resprite

LordVollkorn committed:
The new atom storage broke the emag capability, added a small fix

NanoTrasen-Inc commented:
Been using this for a few rounds. Really nice! Well implemented!

LordVollkorn commented:
> Been using this for a few rounds. Really nice! Well implemented!

Thank you for testing it so diligently 🚀

NanoTrasen-Inc commented:
I do however want to add that the model still shows the lid on it. :P

I know this is just a port though.

-------- Original Message --------
On Jul 15, 2022, 10:21 AM, LordVollkorn wrote:

>> Been using this for a few rounds. Really nice! Well implemented!
>
> Thank you for testing it so diligently 🚀
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/tgstation/tgstation/pull/68193#issuecomment-1185648598), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AANKSMLS4IRBEV2EI3ANCJDVUF6WNANCNFSM52ULG7XA).
> You are receiving this because you commented.Message ID: ***@***.***>
